### PR TITLE
feat(codebases): makes codebase an event 

### PR DESCRIPTION
**Pull request by bartoszmajsak**
Makes codebase an event to propagate its id back to the client

### Checklist

- [x] I followed the contribution [guidelines](https://raw.githubusercontent.com/fabric8-launcher/launcher-backend/master/CONTRIBUTING.md).
- [ ] I created an [issue](https://github.com/fabric8-launcher/launcher-backend/issues/new) and used it in the commit message.
- [x] I have built the project locally prior to submission with `mvn clean install`.
- [x] I kept a clean commit log (no unnecessary commits, no merge commits).
- [x] I am happy with my contribution.

### CODE REVIEW

File: core/core-api/src/main/java/io/fabric8/launcher/core/api/events/StatusEventType.java [hunk 1]
```diff
@@ -10,6 +10,7 @@
     GITHUB_PUSHED("Pushing your customized Booster code into the repo"),
     OPENSHIFT_CREATE("Creating your project on OpenShift"),
     OPENSHIFT_PIPELINE("Setting up your build pipeline"),
+    OSIO_CODEBASE_CREATED("Setting OSIO Codebase"),
```
Comment (bartoszmajsak) [inline core/core-api/src/main/java/io/fabric8/launcher/core/api/events/StatusEventType.java]: @gastaldi I am not happy about mixing OSIO and upstream through events. Do you have any suggestion on how to decouple it?

Comment (gastaldi) [reply core/core-api/src/main/java/io/fabric8/launcher/core/api/events/StatusEventType.java]: Extract the enum into an interface and have separate enums for each mission control implementation?

File: addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/OsioWitClient.java [hunk 2]
```diff
@@ -104,7 +108,18 @@ public void createCodeBase(final String spaceId, final String stackId, final URI
         final Request request = newAuthorizedRequestBuilder(getWitUrl(), "/api/spaces/" + spaceId + "/codebases")
                 .post(create(parse("application/json"), payload))
                 .build();
-        httpClient.executeAndConsume(request, r -> validateCodeBaseResponse(spaceId, repositoryCloneUri, r));
+        httpClient.executeAndConsume(request, r -> {
```
Comment (gastaldi) [inline addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/OsioWitClient.java]: Use `httpClient.executeAndParseJson` instead

File: core/core-api/src/main/java/io/fabric8/launcher/core/api/events/StatusEvent.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package io.fabric8.launcher.core.api.events;
+
+public interface StatusEvent {
```
Comment (gastaldi) [inline core/core-api/src/main/java/io/fabric8/launcher/core/api/events/StatusEvent.java]: A javadoc would be nice

File: addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/OsioWitClient.java (hunk 2 above)
Comment (bartoszmajsak) [reply addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/OsioWitClient.java]: If I want to preserve the logic of `validateCodeBaseResponse` aka `isCodeBaseValid` I can't really do that, can I?

Comment (gastaldi) [reply addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/OsioWitClient.java]: Then use `executeAndMap` if you need the `Response`

File: core/core-api/src/main/java/io/fabric8/launcher/core/api/events/StatusEvent.java (hunk 3 above)
Comment (bartoszmajsak) [reply core/core-api/src/main/java/io/fabric8/launcher/core/api/events/StatusEvent.java]: f0509f3

**Comment by gastaldi** [comment]
WDYT of `LauncherStatusEvent` instead of `LauncherStatusEventType`?

File: addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/OsioWitClient.java (hunk 2 above)
Comment (bartoszmajsak) [reply addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/OsioWitClient.java]: It has a flow anyway, thanks for pointing out - revamping it.

File: core/core-api/src/main/java/io/fabric8/launcher/core/api/events/StatusEvent.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package io.fabric8.launcher.core.api.events;
+
+
+/**
+ * Contract for all status messages that we send to the clients via
+ * websockets to inform them about the status of their project.
+ */
+public interface StatusEvent {
```
Comment (gastaldi) [inline core/core-api/src/main/java/io/fabric8/launcher/core/api/events/StatusEvent.java]: I think it makes sense to name it `StatusEventType`, as it's part of `StatusMessageEvent`

Comment (bartoszmajsak) [reply core/core-api/src/main/java/io/fabric8/launcher/core/api/events/StatusEvent.java]: 9747e0d

**Pull request merged**